src/Raven.Server/Documents/Versioning/VersioningStorage.cs

```diff
@@ -310,15 +310,16 @@ private VersioningConfigurationCollection GetVersioningConfiguration(CollectionN
             if ((nonPersistentFlags & NonPersistentDocumentFlags.FromSmuggler) == NonPersistentDocumentFlags.FromSmuggler)
                 return;
 
-            if (configuration.MaxRevisions.HasValue == false || configuration.MaxRevisions.Value == int.MaxValue)
+            if (configuration.MinimumRevisionsToKeep.HasValue == false &&
+                configuration.MinimumRevisionAgeToKeep.HasValue == false)
                 return;
 
-            var numberOfRevisionsToDelete = revisionsCount - configuration.MaxRevisions.Value;
+            var numberOfRevisionsToDelete = revisionsCount - configuration.MinimumRevisionsToKeep ?? 0;
```

Comment:
No, if the value is zero it means that we going to delete all revisions but than we do not delete if the age criteria is not met.